src/ast/variables/Variable.ts

```diff
@@ -36,7 +36,13 @@ export default class Variable implements ExpressionEntity {
 		_options: ExecutionPathOptions
 	) { }
 
-	getName (): string {
+	getName (reset?: boolean): string {
+		if (reset && this.safeName && this.safeName !== this.name && 
+				this.safeName[this.name.length] === '$' &&
+				this.safeName[this.name.length + 1] === '$') {
+			this.safeName = undefined;
+			return this.name;
+		}
```

Comment:
I see what you mean. I think this is not too bad for now but yes, this is a weird interaction between different runs we should probably fix more substantially at some point. Good catch, though 👍